compiler/resolution/lowerIterators.cpp

```diff
@@ -240,7 +251,7 @@ static bool find_recursive_caller(FnSymbol* fn, FnSymbol* iter, Vec<FnSymbol*>&
   forv_Vec(CallExpr, call, *fn->calledBy)
   {
     // Extract the symbol representing the calling function.
-    FnSymbol* caller = toFnSymbol(call->parentSymbol);
+    FnSymbol* caller = findNonShadowVarParent(call);      
```

Comment:
Shouldn't this just use call->getFunction() instead of call->parentSymbol? I think call->getFunction() would already do the equivalent of findNonShadowVarParent?